### PR TITLE
use new jinja syntax and move OS X to CircleCI

### PR DESCRIPTION
**Pull request by ocefpaf**
@conda-forge-admin, please rerender

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -1,49 +1,15 @@
...
-}
-
-# If an error occurs, run our error handler to output a tail of the build.
-trap 'error_handler' ERR
-
-# Set up a repeating loop to send some output to Travis.
-bash -c "while true; do echo \$(date) - building ...; sleep $PING_SLEEP; done" &
-PING_LOOP_PID=$!
-
-## START BUILD
```
Comment (ocefpaf) [inline recipe/build.sh:28]: We don't need this trick to reduce the log size on CircleCI.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -30,15 +32,10 @@ requirements:
     - pango
     - pixman
     - pyqt
-    - libgcc  # [linux]
-    - libgfortran  # [osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml:34]: Using the jinja syntax for compiler we get this recipe ready for the compiler migration and ensure that the proper runtimes will be install accordingly.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -30,15 +32,10 @@ requirements:
...
 
 test:
   imports:
     - pyferret
-  commands:
-    - conda inspect linkages -p $PREFIX pyferret  # [not win]
-    - conda inspect objects -p $PREFIX pyferret  # [osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml:41]: These are built in tests in conda-build now.

File: conda-forge.yml [hunk 4]
```diff
@@ -4,3 +4,5 @@ travis:
 appveyor:
   secure:
     BINSTAR_TOKEN: ipv/06DzgA7pzz2CIAtbPxZSsphDtF+JFyoWRnXkn3O8j7oRe3rzqj3LOoq2DZp4
+provider:
+  osx: circle
```
Comment (ocefpaf) [inline conda-forge.yml:8]: With this change we use CircleCI for OS X instead of Travis-CI.

File: recipe/build.sh [hunk 5]
```diff
@@ -1,49 +1,15 @@
...
-
 if [ $(uname) == Darwin ]; then
-    export CC="clang"
-    export CXX="clang++"
-    export FC="gfortran"
-    export MACOSX_DEPLOYMENT_TARGET="10.9"
-    export CXXFLAGS="-stdlib=libc++ $CXXFLAGS"
-    export CXXFLAGS="$CXXFLAGS -stdlib=libc++"
-    export DYLD_FALLBACK_LIBRARY_PATH="$PREFIX/lib"
```
Comment (ocefpaf) [inline recipe/build.sh]: Many of these flags are pre-defined now.
Maybe we need to adjust some of those but if we are lucky the OS X build may just work with the defaults.

**Comment by ocefpaf** [comment]
@karlmsmith what is the status for Python 3?

**Comment by karlmsmith** [comment]
Builds I have done for Python 3.4 (RHEL7) and 3.6 (Mac OS X Homebrew) work fine.  Python 3.7 (Mac OS X Homebrew) has some issues with my python objects but everything fine within ferret.  So a Python 3.6 build should work as-is since $PY_VER is used to create the site_specific.mk files.

**Comment by ocefpaf** [comment]
> So a Python 3.6 build should work as-is since $PY_VER is used to create the site_specific.mk files.

Cool, I'll give it a try here and update this PR. Thanks!

**Comment by karlmsmith** [comment]
Error at the beginning of the output:
```
/Users/distiller/miniconda3/conda-bld/pyferret_1533240515778/work/conda_build.sh: line 69: --print-file-name=libgfortran.dylib: command not found
```
This is coming from the line in build.sh indicating the `$FC` is not defined:
```
export GFORTRAN_LIB=`$FC --print-file-name=libgfortran.dylib`
```

**Comment by ocefpaf** [comment]
> This is coming from the line in build.sh indicating the $FC is not defined:

I'll check the toolchain, it should be there :-(

**Comment by ocefpaf** [comment]
I need to investigate this a little bit more, this script is trying to copy system libs:

```
( cd /Users/distiller/miniconda3/conda-bld/pyferret_1533299765631/work; ./copy_dylibs.sh )
./copy_dylibs.sh: line 14: brew: command not found
Copying dylib libraries from /lib
    cairo
cp: /lib/libcairo.*.dylib: No such file or directory
    fontconfig
cp: /lib/libfontconfig.*.dylib: No such file or directory
    freetype
cp: /lib/libfreetype.*.dylib: No such file or directory
    fribidi
cp: /lib/libfribidi.*.dylib: No such file or directory
    glib-2.0
cp: /lib/libglib-2.0.*.dylib: No such file or directory
    gobject-2.0
cp: /lib/libgobject-2.0.*.dylib: No such file or directory
    graphite2
cp: /lib/libgraphite2.*.dylib: No such file or directory
    gthread-2.0
cp: /lib/libgthread-2.0.*.dylib: No such file or directory
    harfbuzz
cp: /lib/libharfbuzz.*.dylib: No such file or directory
    hdf5
cp: /lib/libhdf5.*.dylib: No such file or directory
    hdf5_hl
cp: /lib/libhdf5_hl.*.dylib: No such file or directory
    netcdf
cp: /lib/libnetcdf.*.dylib: No such file or directory
    netcdff
cp: /lib/libnetcdff.*.dylib: No such file or directory
    pango-1.0
cp: /lib/libpango-1.0.*.dylib: No such file or directory
    pangocairo-1.0
cp: /lib/libpangocairo-1.0.*.dylib: No such file or directory
    pangoft2-1.0
cp: /lib/libpangoft2-1.0.*.dylib: No such file or directory
    pcre
cp: /lib/libpcre.*.dylib: No such file or directory
    pixman-1
cp: /lib/libpixman-1.*.dylib: No such file or directory
    png16
cp: /lib/libpng16.*.dylib: No such file or directory
    sz
cp: /lib/libsz.*.dylib: No such file or directory
Copying ffi library from /Cellar/libffi/*/lib
cp: /Cellar/libffi/*/lib/libffi.*.dylib: No such file or directory
Copying intl library from /Cellar/gettext/*/lib
cp: /Cellar/gettext/*/lib/libintl.*.dylib: No such file or directory
Copying dylib libraries from /opt/gcc/lib/gcc/*
    gfortran
cp: /opt/gcc/lib/gcc/*/libgfortran.*.dylib: No such file or directory
    quadmath
cp: /opt/gcc/lib/gcc/*/libquadmath.*.dylib: No such file or directory
    gcc_s
cp: /opt/gcc/lib/gcc/*/libgcc_s.*.dylib: No such file or directory
make: [optimized] Error 1 (ignored)
```

**Comment by karlmsmith** [comment]
That can be ignored (and the build should and I think does).  That was for my build using Homebrew libraries and then copying those Homebrew libraries so they would be included in the distribution tarball.  (In other words, a hack until I learn how to do a proper Mac distribution.)

**Comment by karlmsmith** [comment]
So something different about the fortran compiler on circleci I guess.  I am surprised the fortran compiler does not add the appropriate values for `@rpath`.  May need to include an -rpath in the build for `bin/build_fonts/unix/binary` (which is just used to create the binary font files then is thrown away).  But not sure what the value of -rpath should be.

Just FYI - some info from the `dyld` man page on Mac OS X:
```
@rpath/

Dyld  maintains a current stack of paths called the run path list.  When 
@rpath is encountered it is substituted with each path in the run path list 
until a loadable dylib  if  found.   The run path  stack is built from the 
LC_RPATH load commands in the depencency chain that lead to the current 
dylib load.  You can add an LC_RPATH load command to  an  image  with  
the  -rpath option  to  ld(1).   You  can  even  add  a  LC_RPATH  load  
command  path  that  starts with @loader_path/, and it will push a path 
on the run path stack that relative to the  image containing  the  LC_RPATH. 
The  use  of @rpath is most useful when you have a complex directory 
structure of programs and dylibs which can be installed  anywhere,  
but  keep their  relative positions.  This scenario could be implemented 
using @loader_path, but every client of a dylib could need a different 
load path because its relative position in the file system is  different. 
The  use  of  @rpath introduces a level of indirection that simplifies 
things.  You pick a location in your directory structure as an anchor 
point.  Each dylib then gets an install path that  starts  with @rpath 
and is the path to the dylib relative to the anchor point. Each main 
executable is linked with -rpath @loader_path/zzz, where zzz is 
the path from  the  executable to the anchor point.  At runtime dyld 
sets its run path to be the anchor point, then each dylib is found 
relative to the anchor point.
```

**Comment by karlmsmith** [comment]
Doh!  Just realized the solution to this is to add a `--static-libgfortran` flag to the fortran compiler flags.  Should have no affect on compiling source to objects.
(The reason for the `GFORTRAN_LIB` definition is linking through setup.py which does not know to add the gfortran library.)

**Comment by karlmsmith** [comment]
The `platform_specific.mk.*` files assign (do not add to) `CFLAGS` and `FFLAGS`.  The appropriate one of these is added after `site_specific.mk`, so any settings of these flags even in that file will even be ignored.  I should change this for future version, but for now it probably needs a patch to `platform_specific.mk.intel-mac` to change
```
        FFLAGS          = -fPIC \
```
to
```
        FFLAGS          = -fPIC -static-libgfortran \
```
(or whatever flags you want to try).  Or maybe instead change the `=` to `+=`

**Pull request merged**